src/transformers/modeling_attn_mask_utils.py

```diff
@@ -234,6 +234,59 @@ def _unmask_unattended(
 
         return expanded_mask.mul(~torch.all(expanded_mask == min_dtype, dim=-1, keepdim=True))
 
+    @staticmethod
+    def _ignore_causal_mask_sdpa(
```

Comment:
I wonder if it is reasonable to have this part of logic in sdpa kernel.

Reply:
I think we only want to do this once per model forward, not once every sdpa attention layer.

Reply:
Kind of the same thing with FA2 backend & avoiding unpad/pad at each SDPA call, currently this is possible only with the NestedTensor implem but transformers do not use them https://github.com/huggingface/transformers/issues/29983